test/quantization/test_backward_compatibility.py

```diff
@@ -69,6 +71,69 @@ def remove_prefix(text, prefix):
         self.assertEqual(qmodule_scripted(input_tensor), expected, atol=prec)
         self.assertEqual(qmodule_traced(input_tensor), expected, atol=prec)
 
+    def _test_op_graph(self, qmodule, subname=None, input_size=None, input_quantized=True,
+                       generate=False, prec=None, new_zipfile_serialization=False):
+        r"""
+        Input: a floating point module
+
+        If generate == True, traces and scripts the module and quantizes the results with
+        PTQ, and saves the results.
+
+        If generate == False, traces and scripts the module and quantizes the results with
+        PTQ, and compares to saved results.
+        """
+        # NB: we take __file__ from the module that defined the test
+        # class, so we place the expect directory where the test script
+        # lives, NOT where test/common_utils.py lives.
+        module_id = self.__class__.__module__
+        munged_id = remove_prefix(self.id(), module_id + ".")
+        test_file = os.path.realpath(sys.modules[module_id].__file__)
+        base_name = os.path.join(os.path.dirname(test_file),
+                                 "serialized",
+                                 munged_id)
+
+        subname_output = ""
+        if subname:
+            base_name += "_" + subname
+            subname_output = " ({})".format(subname)
+
+        input_file = base_name + ".input.pt"
+        state_dict_file = base_name + ".state_dict.pt"
+        scripted_module_file = base_name + ".scripted.pt"
+        traced_module_file = base_name + ".traced.pt"
+        expected_file = base_name + ".expected.pt"
+
+        # only generate once.
+        if generate and qengine_is_fbgemm():
+            input_tensor = torch.rand(*input_size).float()
+            torch.save(input_tensor, input_file)
+
+            # convert to TorchScript
+            scripted = torch.jit.script(qmodule)
+            traced = torch.jit.trace(qmodule, input_tensor)
+
+            # quantize
+
+            def _eval_fn(model, data):
+                model(data)
+
+            qconfig_dict = {'': torch.quantization.default_qconfig}
+            scripted_q = torch.quantization.quantize_jit(
+                scripted, qconfig_dict, _eval_fn, [input_tensor])
+            traced_q = torch.quantization.quantize_jit(
+                traced, qconfig_dict, _eval_fn, [input_tensor])
+
+            torch.jit.save(scripted_q, scripted_module_file)
+            torch.jit.save(traced_q, traced_module_file)
+            torch.save(scripted_q(input_tensor), expected_file)
+
+        input_tensor = torch.load(input_file)
+        qmodule_scripted = torch.jit.load(scripted_module_file)
+        qmodule_traced = torch.jit.load(traced_module_file)
+        expected = torch.load(expected_file)
+        self.assertEqual(qmodule_scripted(input_tensor), expected, atol=prec)
+        self.assertEqual(qmodule_traced(input_tensor), expected, atol=prec)
```

Comment:
for potentially reusing parts of this, it's different enough that I think the current way is cleaner given the readability benefits